lib/actions/api.js

```diff
@@ -45,22 +49,75 @@ export function planTrip (customOtpQueryBuilder) {
   }
 }
 
+// Profile Query
+
+export const receivedProfileError = createAction('PROFILE_ERROR')
+export const receivedProfileResponse = createAction('PROFILE_RESPONSE')
+export const requestProfileResponse = createAction('PROFILE_REQUEST')
+
+function constructPlanQuery (api, query) {
+  const planEndpoint = `${api.host}${api.port ? ':' + api.port : ''}${api.path}/plan`
+
+  let params = {}
+  queryParams.filter(qp => qp.planTypes.indexOf('ITINERARY') !== -1 && qp.name in query).forEach(qp => {
+    params = Object.assign(params, qp.itineraryRewrite
+      ? qp.itineraryRewrite(query[qp.name])
+      : { [qp.name]: query[qp.name] })
```

Comment:
I think it would be good to run some kind of validation on each param that is sent to make sure that it is the right type.